Mage.Sets/src/mage/cards/b/BaneclawMarauder.java

```diff
@@ -0,0 +1,127 @@
+package mage.cards.b;
+
+import mage.MageInt;
+import mage.MageObjectReference;
+import mage.abilities.common.BecomesBlockedSourceTriggeredAbility;
+import mage.abilities.common.DiesCreatureTriggeredAbility;
+import mage.abilities.effects.common.LoseLifeTargetControllerEffect;
+import mage.abilities.effects.common.continuous.BoostAllEffect;
+import mage.abilities.keyword.NightboundAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Duration;
+import mage.constants.SubType;
+import mage.constants.WatcherScope;
+import mage.filter.FilterPermanent;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.ObjectSourcePlayer;
+import mage.filter.predicate.ObjectSourcePlayerPredicate;
+import mage.filter.predicate.permanent.BlockingOrBlockedBySourcePredicate;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.watchers.Watcher;
+
+import java.util.*;
+
+/**
+ * @author TheElk801
+ */
+public final class BaneclawMarauder extends CardImpl {
+
+    private static final FilterCreaturePermanent filter
+            = new FilterCreaturePermanent("each creature blocking it");
+    private static final FilterPermanent filter2 = new FilterCreaturePermanent("a creature blocking {this}");
+
+    static {
+        filter.add(BlockingOrBlockedBySourcePredicate.BLOCKING);
+        filter2.add(BaneclawMarauderPredicate.instance);
+    }
+
+    public BaneclawMarauder(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "");
+
+        this.subtype.add(SubType.WEREWOLF);
+        this.power = new MageInt(5);
+        this.toughness = new MageInt(4);
+        this.color.setBlack(true);
+        this.nightCard = true;
+
+        // Whenever Baneclaw Marauder becomes blocked, each creature blocking it gets -1/-1 until end of turn.
+        this.addAbility(new BecomesBlockedSourceTriggeredAbility(new BoostAllEffect(
+                -1, -1, Duration.EndOfTurn, filter, false
+        ), false));
+
+        // Whenever a creature blocking Baneclaw Marauder dies, its controller loses 1 life.
+        this.addAbility(new DiesCreatureTriggeredAbility(
+                new LoseLifeTargetControllerEffect(1)
+                        .setText("its controller loses 1 life"),
+                false, filter2, true
+        ), new BaneclawMarauderWatcher());
```

Comment:
Out of curiosity, why did you choose to use `DiesCreatureTriggeredAbility`, `LoseLifeTargetControllerEffect` and a watcher rather than `ZoneChangeEvent` and `LoseLifePermanentControllerEffect`?

Reply:
The short answer is because right now effects which care about attacking/blocking creatures dying have some issues and need to be refactored. `BlockingAttackerIdPredicate` doesn't work if they ability is copied as it has the source object's id tied to it. Also `LoseLifePermanentControllerEffect` causes the controller of the source permanent to lose life, which isn't what we need.

Reply:
Yes, the (sourceObject) issue with "copies" is rampant in regards to many of these abilities.

Reply:
Make issues with that bug and examples. Copy code must be fixed.

Reply:
just opened #8371